### DISCUSSION TITLE
Problem with load_project() function

### DISCUSSION THREAD

**Discussion by DMaxJ** [PMEAL/OpenPNM#2397] [Q&A]
I am running some simulations. To save time required for extracting the network every time I run a simulation I tried saving the project with 
`ws.save_project(proj)`
which creates the desired .pnm file with the project name.

When I later try to open the same, I get some error. I could use the same functions in the earlier versions of openpnm. Now, I get error when I use 
`load_project('myproject')`
the details of which I have provided below
```
ws.load_project('myproject')

Traceback (most recent call last):


 Input In [55] in <cell line: 1>
    ws.load_project('myproject')

  File c:\users\XYZ\documents\porenetwork\openpnm\utils\_workspace.py:223 in load_project
    proj = PNM.load_project(filename=filename)

  File c:\users\XYZ\documents\porenetwork\openpnm\io\_pnm.py:94 in load_project
    proj, obj = create_obj(root, name, proj)

  File c:\users\XYZ\documents\porenetwork\openpnm\io\_pnm.py:128 in create_obj
    obj.models.update(json.loads(root[name].attrs['models']))

  File c:\users\XYZ\documents\porenetwork\openpnm\core\_models.py:28 in update
    target = self._find_parent()

  File c:\users\XYZ\documents\porenetwork\openpnm\core\_models.py:41 in _find_parent
    raise Exception("No parent object found!")

Exception: No parent object found!

```

All my scripts and files are inside the same directory. There is no issue when I export files to VTK format. While saving an object, project, or workspace I get no error message. Could you please help solve this problem?

**Comment by jgostick** [comment]
The best work around for now, since we may not be able to look into this right away, is to save the ``dict`` coming out of the ``snow2`` algorithm using python's ``pickle`` module.  This dict is just a pure python object with no complicated dependencies and will always just work.  Loading the dict into openpnm is not the slow part right?

**Reply by DMaxJ** [reply]
Thanks. I will check that. Loading the dict is definitely not the problem!!

**Reply by jgostick** [reply]
We have quite a few tests for the save/load functionality, so on the one hand I'm surprized that it's breaking, but on the other hand, it's actually pretty complicated under the hood.

**Reply by DMaxJ** [reply]
I actually tried going to the implemented functions under the hood, after after spending a couple of hours I gave up. I tried to read the file running the functions line by line (as implemented in the source code. 
`obj.models.update(json.loads(root[name].attrs['models']))`
The saved .pnm file actually has 'models' but I could not figure out why the 
`_find_parent(self):`
fails.

**Reply by jgostick** [reply]
FWIW, the "interconnectedness" of objects in OpenPNM is a HUGE pain, leading to problems like the on you're having...and we have decided to strip it ALL OUT in V3.  This will delay the release of V3, but will be a major improvement in terms of maintainability and hopefully usability too.
